app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -270,21 +256,21 @@ def set_organization_id
       params[:organization_id] = params[:owner] if params[:owner]
     end
 
-    def find_system(uuid = nil)
-      @system = System.where(:uuid => uuid || params[:id]).first
-      if @system.nil?
+    def find_host(uuid = nil)
+      aspect = Katello::Host::SubscriptionAspect.where(:uuid => uuid || params[:id]).first
+      if aspect.nil?
         # check with candlepin if consumer is Gone, raises RestClient::Gone
         Resources::Candlepin::Consumer.get params[:id]
         fail HttpErrors::NotFound, _("Couldn't find consumer '%s'") % params[:id]
```

Comment:
If someone passed `uuid` to this function, this wouldn't work? I know this was already in place, but might consider updating the logic.